[CI] DataStreamIndexSettingsProviderTests testGetAdditionalIndexSettingsNoMappings failing

**Build Scans:**
- [elasticsearch-intake #37413 / part1](https://gradle-enterprise.elastic.co/s/kh6br6zow7naw)
- [elasticsearch-pull-request #136399 / part-1](https://gradle-enterprise.elastic.co/s/lmyltdtw6fa2m)
- [elasticsearch-pull-request #136399 / part-1](https://gradle-enterprise.elastic.co/s/lxmv7frvrxj7w)
- [elasticsearch-pull-request #135490 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/luobvlpyaawce)
- [elasticsearch-pull-request #135490 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/cqbjsw6h72hpc)
- [elasticsearch-pull-request #135467 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/umqb3gwokr2cq)
- [elasticsearch-pull-request #135467 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/iit2n2l526hve)
- [elasticsearch-pull-request #135435 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/vceyqa7sfwcge)
- [elasticsearch-pull-request #135435 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/uojlbtizjdato)
- [elasticsearch-pull-request #135432 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/3qd3zzqbug6d4)

**Reproduction Line:**
```
./gradlew ":modules:data-streams:test" --tests "org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests.testGetAdditionalIndexSettingsNoMappings" -Dtests.seed=18E619D8D5829A98 -Dtests.locale=en-AT -Dtests.timezone=America/Managua -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testGetAdditionalIndexSettingsNoMappings'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <3>
     but: was <4>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testGetAdditionalIndexSettingsNoMappings
- [main] 1 consecutive failures in step part1
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 10 failures in test testGetAdditionalIndexSettingsNoMappings (1.1% fail rate in 922 executions)
- [main] 2 failures in step part-1 (0.3% fail rate in 581 executions)
- [main] 7 failures in step checkpart1_release-tests (33.3% fail rate in 21 executions)
- [main] 5 failures in pipeline elasticsearch-pull-request (0.9% fail rate in 571 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/data-streams/src/test/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProviderTests.java
+++ b/modules/data-streams/src/test/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProviderTests.java
@@ -53,6 +53,7 @@ public class DataStreamIndexSettingsProviderTests extends ESTestCase {
     private boolean expectedIndexDimensionsTsidOptimizationEnabled;
     private IndexVersion indexVersion;
     private boolean expectedDisabledSequenceNumbers;
+    private boolean expectedSyntheticId;
 
     @Before
     public void setup() {
@@ -67,14 +68,23 @@ public class DataStreamIndexSettingsProviderTests extends ESTestCase {
             indexVersion = IndexVersionUtils.randomPreviousCompatibleVersion(IndexVersions.TSID_CREATED_DURING_ROUTING);
         }
         expectedDisabledSequenceNumbers = indexVersion.onOrAfter(IndexVersions.TIME_SERIES_DISABLE_SEQUENCE_NUMBERS_DEFAULT);
+        expectedSyntheticId = indexVersion.onOrAfter(IndexVersions.TIME_SERIES_USE_SYNTHETIC_ID_DEFAULT_PROD);
         indexDimensionsTsidStrategyEnabledSetting = usually();
         expectedIndexDimensionsTsidOptimizationEnabled = indexDimensionsTsidStrategyEnabledSetting
             && indexVersion.onOrAfter(IndexVersions.TSID_CREATED_DURING_ROUTING);
     }
 
     int maybeAdjustIndexSettingCount(int baseCount) {
-        // We need to adjust to account for the seq_no removal and synthetic id settings
-        return expectedDisabledSequenceNumbers ? baseCount + 2 : baseCount;
+        // Adjust count independently: DISABLE_SEQUENCE_NUMBERS and SYNTHETIC_ID use different version thresholds
+        // (TIME_SERIES_DISABLE_SEQUENCE_NUMBERS_DEFAULT and TIME_SERIES_USE_SYNTHETIC_ID_DEFAULT_PROD respectively)
+        int count = baseCount;
+        if (expectedDisabledSequenceNumbers) {
+            count++;
+        }
+        if (expectedSyntheticId) {
+            count++;
+        }
+        return count;
     }
 
     public void testGetAdditionalIndexSettings() throws Exception {

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -268,15 +268,6 @@ tests:
 - class: org.elasticsearch.xpack.entsearch.EnterpriseSearchRestIT
   method: test {p0=entsearch/rules/40_rule_query_search/Perform a rule query with an organic query that must be rewritten to another query type}
   issue: https://github.com/elastic/elasticsearch/issues/146106
-- class: org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests
-  method: testGetAdditionalIndexSettingsNoMappings
-  issue: https://github.com/elastic/elasticsearch/issues/146108
-- class: org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests
-  method: testGetAdditionalIndexSettingsIndexRoutingPathAlreadyDefined
-  issue: https://github.com/elastic/elasticsearch/issues/146109
-- class: org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests
-  method: testGetAdditionalIndexSettingsMigrateToTsdb
-  issue: https://github.com/elastic/elasticsearch/issues/146112
 - class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
   method: test {csv-spec:embedding.embedding with multiple knn queries in fork}
   issue: https://github.com/elastic/elasticsearch/issues/146130
@@ -286,12 +277,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT
   method: test {csv-spec:embedding.embedding with multiple knn queries in fork}
   issue: https://github.com/elastic/elasticsearch/issues/146137
-- class: org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests
-  method: testGenerateRoutingPathFromDynamicTemplate
-  issue: https://github.com/elastic/elasticsearch/issues/146197
-- class: org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests
-  method: testGetAdditionalIndexSettingsMappingsMerging
-  issue: https://github.com/elastic/elasticsearch/issues/146198
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=tsdb/25_id_generation/*}
   issue: https://github.com/elastic/elasticsearch/issues/146177
@@ -313,9 +298,6 @@ tests:
 - class: org.elasticsearch.cluster.routing.allocation.decider.WriteLoadConstraintDeciderIT
   method: testMaxQueueLatencyMetricIsPublished
   issue: https://github.com/elastic/elasticsearch/issues/146283
-- class: org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests
-  method: testGetAdditionalIndexSettingsLookAheadTime
-  issue: https://github.com/elastic/elasticsearch/issues/146305
 - class: org.elasticsearch.xpack.sql.qa.multi_node.RestSqlIT
   method: testAsyncTextPaginated
   issue: https://github.com/elastic/elasticsearch/issues/80089
@@ -328,11 +310,6 @@ tests:
 - class: org.elasticsearch.xpack.sql.qa.multi_cluster_with_security.RestSqlIT
   method: testAsyncTextPaginated
   issue: https://github.com/elastic/elasticsearch/issues/80089
-- class: org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests
-  method: testGetAdditionalIndexSettingsLookBackTime
-  issue: https://github.com/elastic/elasticsearch/issues/146307
-- class: org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests
-  issue: https://github.com/elastic/elasticsearch/issues/146126
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=indices.resolve_cluster/20_resolve_system_index/Resolve system index via resolve/cluster}
   issue: https://github.com/elastic/elasticsearch/issues/146312

